future-apps/payroll/contracts/Payroll.sol

```diff
@@ -634,4 +667,79 @@ contract Payroll is EtherTokenConstant, IForwarder, IsContract, AragonApp {
 
         return xrt;
     }
+
+    /**
+     * @dev Loops over tokens to send requested amount to the employee
+     * @param _employeeId Employee's identifier
+     * @param _totalAmount Total amount to be transferred to the employee distributed through the setup tokens allocation.
+     * @param _reference String detailing payment reason.
+     * @return True if there was at least one token transfer.
+     */
+    function _transferTokensAmount(uint256 _employeeId, uint256 _totalAmount, string _reference) private returns (bool somethingPaid) {
+        Employee storage employee = employees[_employeeId];
+        for (uint256 i = 0; i < allowedTokensArray.length; i++) {
+            address token = allowedTokensArray[i];
+            if (employee.allocation[token] != uint256(0)) {
+                uint128 exchangeRate = _getExchangeRate(token);
+                require(exchangeRate > 0, ERROR_EXCHANGE_RATE_ZERO);
+                // Salary converted to token and applied allocation percentage
+                uint256 tokenAmount = _totalAmount.mul(exchangeRate).mul(employee.allocation[token]);
+                // Divide by 100 for the allocation and by ONE for the exchange rate
+                tokenAmount = tokenAmount / (100 * ONE);
+                finance.newPayment(token, employee.accountAddress, tokenAmount, 0, 0, 1, _reference);
+                emit SendPayment(employee.accountAddress, token, tokenAmount, _reference);
+                somethingPaid = true;
+            }
+        }
+    }
+
+    /**
+     * @dev Tries removing employee if there are no pending payments and has reached employee's end date.
+     * @param _employeeId Employee's identifier
+     */
+    function _tryRemovingEmployee(uint256 _employeeId) private {
```

Comment:
I think we discussed this already, but of course I don't remember, haha. I think the reason was precisely because we don't have the "re-enabling" functionality and the address would be blocked. This way an employee can  be re-enabled by creating a new one and using the same address. Does it make sense?

Reply:
👍 In that case, it seems like we have two paths:

1. Introduce "re-enabling" functionality for terminated employees in the future, and avoid storage gaps now
1. Use "re-adding" someone as a new employee as the way to "re-hire" someone, and never allow terminated employees to be re-enabled with the same ID

Not sure which one is the best real world analogy, but I am also starting to get the sense that assigning a new employee ID is not so bad for rehires.

Reply:
I think that's what we are doing currently, see [here](https://github.com/aragon/aragon-apps/pull/744/commits/c1b8831675d81158ab7072f2aa0c741619583043#diff-c669e10c65bc3c52e31525560eae501bR96).